src/leaspy/io/logs/fit_output_manager.py

```diff
@@ -159,41 +162,145 @@ def save_model_parameters_convergence(
-        if getattr(model, "is_ordinal", False):
+        to_skip = {"betas", "sources", "space_shifts", "xi", "tau", "xi_mean"}
+        if model.name == "ordinal":
             to_skip.add("deltas")
-        params_to_plot = model.state.tracked_variables - to_skip
-
-        n_plots = len(params_to_plot)
+        params_with_feature_labels = ["g", "v0"]
+        params_with_sources = ["mixing_matrix"]
+        params_with_events = []
+        if model.name == "joint":
+            to_skip.add("survival_shifts")
+            params_with_sources.append("zeta")
+            params_with_events += ["nu", "rho"]
+
+        params_to_plot = list(model.state.tracked_variables - to_skip)
+
+        files = os.listdir(self.path_save_model_parameters_convergence)
+        files_to_plot = [
+            file
+            for file in files
+            if any(file.startswith(param) for param in params_to_plot)
+        ]
+        # To plot related parameters close to each other, we sort the list
+        files_to_plot.sort()
+
+        n_plots = len(files_to_plot)
         n_rows = math.ceil(n_plots / 2)
-        _, ax = plt.subplots(n_rows, 2, figsize=(width, n_rows * height_per_row))
-
-        for i, parameter_name in enumerate(params_to_plot):
-            df_convergence = pd.read_csv(
-                self.path_save_model_parameters_convergence / f"{parameter_name}.csv",
-                index_col=0,
-                header=None,
-            )
-            df_convergence.index.rename("iter", inplace=True)
 
-            x_position = i // 2
-            y_position = i % 2
-            df_convergence.plot(ax=ax[x_position][y_position], legend=False)
-            ax[x_position][y_position].set_title(parameter_name)
-
-        plt.tight_layout()
-        plt.savefig(self.path_plot_convergence_model_parameters)
-        plt.close()
+        # If plot sourcewise is true, new sourcewise csv files will be created
+        if self.plot_sourcewise:
+            new_files = []
+            for param_name in params_with_sources:
+                related_files = [
+                    file for file in files_to_plot if file.startswith(param_name)
+                ]
+                if not related_files:
+                    continue
+                related_files.sort()
+
+                num_sources = model.source_dimension
+
+                for source_idx in range(num_sources):
+                    combined_data = []
+
+                    for file_name in related_files:
+                        file_path = (
+                            self.path_save_model_parameters_convergence / file_name
+                        )
+                        df = pd.read_csv(file_path, index_col=0, header=None)
+
+                        combined_data.append(df.iloc[:, source_idx])
+
+                    combined_df = pd.concat(combined_data, axis=1, join="inner")
+                    new_file_name = "sourcewise_" f"{param_name}_{source_idx + 1}.csv"
```

Comment:
This looks weird, is this what you wanted to write:

```suggestion
                    new_file_name = f"sourcewise_{param_name}_{source_idx + 1}.csv"
```

Reply:
thanks, since I added `sourcewise` later, I forgot to change "f". now , it is ok